src/analysis/normed_space/star/multiplier.lean

```diff
@@ -0,0 +1,450 @@
+/-
+Copyright (c) 2022 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux, Jon Bannon
+-/
+
+import algebra.star.star_alg_hom
+import analysis.normed_space.star.basic
+import analysis.normed_space.operator_norm
+import analysis.special_functions.pow
+import analysis.normed_space.star.mul
+
+/-!
+# Multiplier Algebra of a C⋆-algebra
+
+Define the multiplier algebra of a C⋆-algebra as the algebra (over `𝕜`) of double centralizers,
+for which we provide the localized notation `𝓜(𝕜, A)`.  A double centralizer is a pair of
+continuous linear maps `L R : A →L[𝕜] A` satisfying the intertwining condition `R x * y = x * L y`.
+
+There is a natural embedding `A → 𝓜(𝕜, A)` which sends `a : A` to the continuous linear maps
+`L R : A →L[𝕜] A` given by left and right multiplication by `a`, and we provide this map as a
+coercion.
+
+The multiplier algebra corresponds to a non-commutative Stone–Čech compactification in the sense
+that when the algebra `A` is commutative, it can be identified with `C₀(X, ℂ)` for some locally
+compact Hausdorff space `X`, and in that case `𝓜(𝕜, A)` can be identified with `C(β X, ℂ)`.
+
+## Implementation notes
+
+We make the hypotheses on `𝕜` as weak as possible so that, in particular, this construction works
+for both `𝕜 = ℝ` and `𝕜 = ℂ`.
+
+The reader familiar with C⋆-algebra theory may recognize that one
+only needs `L` and `R` to be functions instead of continuous linear maps, at least when `A` is a
+C⋆-algebra. Our intention is simply to eventually provide a constructor for this situation.
+
+## TODO
+
++ define a type synonym for `𝓜(𝕜, A)` which is equipped with the strict uniform space structure
+  and show it is complete
++ show that the image of `A` in `𝓜(𝕜, A)` is an essential ideal
++ prove the universal property of `𝓜(𝕜, A)`
+* Construct a double centralizer from a pair of maps (not necessarily linear or continuous)
+  `L : A → A`, `R : A → A` satisfying the centrality condition `∀ x y, R x * y = x * L y`.
+-/
+
+noncomputable theory
+
+open_locale nnreal ennreal
+open nnreal continuous_linear_map
+
+universes u v
+
+/-- The type of *double centralizers*, also known as the *multiplier algebra* and denoted by
+`𝓜(𝕜, A)`, of a non-unital normed algebra. -/
```

Comment:
Perhaps for clarity:
```suggestion
`𝓜(𝕜, A)`, of a non-unital normed algebra.

If `x : 𝓜(𝕜, A)`, then `x.1` and `x.2` are what is usually referred to as $L$ and $R$. -/
```

Reply:
Hmm, is that true? Surely it's rather
```suggestion
`𝓜(𝕜, A)`, of a non-unital normed algebra.

If `x : 𝓜(𝕜, A)`, then `x.fst` and `x.snd` are what is usually referred to as $L$ and $R$. -/
```

Reply:
Actually I see Jireh already wrote the correct thing.